packages/stores/src/queries-external/position-performance/index.ts

```diff
@@ -115,46 +115,38 @@ export class ObservableQueryPositionPerformanceMetrics extends ObservableQueryEx
   }
 
   readonly calculateReturnOnInvestment = computedFn(
-    (currentPositionCoins: CoinPretty[]): RatePretty => {
-      if (!this.response) return new RatePretty(0);
+    (
+      currentPositionCoins: CoinPretty[],
+      unclaimedRewards: CoinPretty[]
+    ): RatePretty => {
+      if (!this.response || this.totalPrincipalValue.toDec().isZero())
+        return new RatePretty(0);
+
+      const currentPositionValue = currentPositionCoins.reduce(
+        (sum, coin) =>
+          sum.add(
+            this.priceStore.calculatePrice(coin) ??
+              new PricePretty(this.fiatCurrency, 0)
+          ),
+        new PricePretty(this.fiatCurrency, 0)
+      );
+      const unclamiedRewardsValue = unclaimedRewards.reduce(
+        (sum, coin) =>
+          sum.add(
+            this.priceStore.calculatePrice(coin) ??
+              new PricePretty(this.fiatCurrency, 0)
+          ),
+        new PricePretty(this.fiatCurrency, 0)
+      );
 
-      // aggregate principal coins by denom
-      const principalCoinDenomMap = new Map<string, CoinPretty>();
-      this.principal.forEach(({ coin }) => {
-        const existingCoin = principalCoinDenomMap.get(
-          coin.currency.coinMinimalDenom
-        );
-        if (existingCoin) {
-          principalCoinDenomMap.set(
-            coin.currency.coinMinimalDenom,
-            existingCoin.add(coin)
-          );
-        } else {
-          principalCoinDenomMap.set(coin.currency.coinMinimalDenom, coin);
-        }
-      });
-
-      // calculate ROI per given coin denom
-      const roiPerCoinDenom = new Map<string, RatePretty>();
-      currentPositionCoins.forEach((coin) => {
-        const denom = coin.currency.coinMinimalDenom;
-        const principalCoin = principalCoinDenomMap.get(denom);
-        if (principalCoin && !principalCoin.toDec().isZero()) {
-          // roi = (finalValue - initialInvestment) / initialInvestment
-          const roi = coin
-            .toDec()
-            .sub(principalCoin.toDec())
-            .quo(principalCoin.toDec());
-          roiPerCoinDenom.set(denom, new RatePretty(roi));
-        }
-      });
-
-      // return the average of all ROI per coin denom
-      const roiPerCoinDenomArray = Array.from(roiPerCoinDenom.values());
-      if (roiPerCoinDenomArray.length === 0) return new RatePretty(0);
-      return roiPerCoinDenomArray
-        .reduce((sum, roi) => sum.add(roi), new RatePretty(0))
-        .quo(new Dec(roiPerCoinDenomArray.length));
+      return new RatePretty(
+        currentPositionValue
+          .toDec()
+          .add(unclamiedRewardsValue.toDec())
+          .add(this.totalEarnedValue.toDec())
+          .sub(this.totalPrincipalValue.toDec())
+          .quo(this.totalPrincipalValue.toDec())
+      );
```

Comment:
@sunnya97 @jonator Can you review this changes that showed up on this PR after rebasing from stage 🙏

Reply:
Yep we changed how we calculate ROI, using $ value instead of token amounts